pkg/http/api.go

```diff
@@ -591,8 +592,14 @@ func (a *api) onDirectMessage(reqCtx *fasthttp.RequestCtx) {
 	resp, err := a.directMessaging.Invoke(reqCtx, targetID, req)
 	// err does not represent user application response
 	if err != nil {
+		// Allowlists policies that are applied on the callee side can return a Permission Denied error.
+		// For everything else, treat it as a gRPC transport error
+		statusCode := fasthttp.StatusInternalServerError
+		if status.Code(err) == codes.PermissionDenied {
+			statusCode = invokev1.HTTPStatusFromCode(codes.PermissionDenied)
```

Comment:
@wcs1only This is a change that relates to https://github.com/dapr/dapr/issues/2338

Reply:
Noted. If you merge before me, I'll be sure be sure to take a look.

Reply:
Thanks!